src/ui/features/onboarding/Success/ReferralCodeWidget/ReferralCodeWidget.tsx

```diff
@@ -0,0 +1,181 @@
+          }}
+          onClick={onClick}
+        >
+          <HStack gap={8}>
+            <GiftIcon />
+            <GradientText>
+              <UIText kind="body/accent">Enter Referral Code</UIText>
+            </GradientText>
+          </HStack>
+        </Button>
+        <UIText
+          kind="small/regular"
+          color="var(--always-white)"
+          style={{ opacity: 0.5 }}
+        >
+          if you have one
+        </UIText>
+      </VStack>
+    </>
+  );
+}
+
+function ReferralCodeForm({
+  onSuccess,
+}: {
+  onSuccess: (pendingReferrer: ReferrerData) => void;
+}) {
+  const [referralCode, setReferralCode] = useState('');
+
+  const { mutate: applyReferralCode, ...applyReferralCodeMutation } =
+    useMutation({
+      mutationFn: ({
+        referralCode: rawReferralCode,
+      }: {
+        referralCode: string;
+      }) => {
+        return walletPort.request('uiApplyReferralCodeToAllWallets', {
+          referralCode: rawReferralCode.trim(),
+        });
+      },
+      onSuccess,
+    });
+
+  return (
+    <>
+      <Spacer height={28} />
+      <VStack gap={16} className={styles.referralCodeForm}>
+        <UIText
+          kind="body/accent"
+          color="var(--always-white)"
+          style={{ textAlign: 'left' }}
+        >
+          Referral Code
+        </UIText>
+        <form
+          onSubmit={(event) => {
+            event.preventDefault();
+
+            const form = event.currentTarget;
+            if (!form.checkValidity()) {
+              return;
+            }
+            const formData = collectData(form, {});
+            const referralCode = formData.referralCode as string;
+
+            applyReferralCode({ referralCode });
+          }}
+        >
+          <VStack gap={4}>
+            <HStack gap={8} alignItems="center">
+              <ZStack>
+                <Input
+                  className={styles.referralCodeInput}
+                  name="referralCode"
+                  autoFocus={true}
+                  boxHeight={40}
+                  placeholder="Enter code"
+                  required={true}
+                  value={referralCode}
+                  onChange={(event) => setReferralCode(event.target.value)}
+                  disabled={applyReferralCodeMutation.isLoading}
+                />
+                {referralCode.length > 0 ? (
+                  <ClearSolidIcon
+                    style={{
+                      width: 20,
+                      height: 20,
+                      cursor: 'pointer',
+                      alignSelf: 'center',
+                      justifySelf: 'end',
+                      color: 'var(--neutral-500)',
+                      marginRight: 8,
+                    }}
+                    onClick={() => {
+                      setReferralCode('');
+                      applyReferralCodeMutation.reset();
+                    }}
+                  />
+                ) : null}
```

Comment:
This must be wrapped in `<UnstyledButton type="button" />` and `onClick` should be on button

Reply:
`onOnclick` on `svg`, what was I even thinking... updated ✅